src/components/main-form.js

```diff
@@ -108,9 +117,9 @@ class MainForm extends React.Component {
     o1.fv = this.calculateFV(o1, o2, state);
     o2.fv = this.calculateFV(o2, o1, state);
 
-    state.optCost = this.calculateOptCost(o1, o2);
+    state.optCost = this.calculateOpportunityCost(o1, o2);
 
-    this.setState(state);
+    return state;
```

Comment:
You can be more of a purist and make a state deep copy, and return a new copy rather than doing modifications :P

Reply:
I could.  If a deep copy is necessary here shouldn't I go back to using a deep copy in updateInput() too?